pkg/yamltemplate/template.go

```diff
@@ -34,6 +34,33 @@ type TemplateOpts struct {
 	ImplicitMapKeyOverrides bool
 }
 
+func IsTemplate(val interface{}) bool {
```

Comment:
it's a little weird to have `val interface{}`. may be have input as DocumentSet and have private isTemplate() function with interface? wdyt?

Reply:
Yeah... I started with that exact structure... but it seemed a bit silly to have an exported function that's a one-liner that calls the unexported function that had all the logic.

Then on top, the first thing we do in the function is downcast that quietly terminates.

From a "when in Rome" perspective, in every other case I saw something like that, here, there was some kind of pre-condition or post-condition work that felt like it justified the separate functions of the same name. 

Perhaps given that it's an exported function is sufficient justification for having as typed of a signature as possible. I'll switch it back. Thanks for asking.